tests/integration/kubernetes/k8s-guest-pull-image.bats

```diff
@@ -59,37 +60,59 @@ setup() {
     assert_rootfs_count "$node" "$sandbox_id" "0"
 }
 
-@test "Test we can pull a large image inside the guest" {
-    [[ " ${SUPPORTED_NON_TEE_HYPERVISORS} " =~ " ${KATA_HYPERVISOR} " ]] && skip "Test not supported for ${KATA_HYPERVISOR}."
-    skip "This test requires large memory, which the encrypted memory is typically small and valuable in TEE. \
-          The test will be skiped until https://github.com/kata-containers/kata-containers/issues/8142 is addressed."
-    kata_pod_with_nydus_config="$(new_pod_config "$large_image" "kata-${KATA_HYPERVISOR}")"
-    set_node "$kata_pod_with_nydus_config" "$node"
-    set_container_command "$kata_pod_with_nydus_config" "0" "sleep" "30"
+@test "Test we can pull a large image inside the guest using trusted ephemeral storage" {
+    
+    # The image pulled in the guest will be downloaded and unpacked in the `/run/kata-containers/image` directory. 
+    # However, by default, systemd allocates 10% of the available physical RAM to the `/run` directory using a `tmpfs` filesystem. 
+    # It means that if we run a kata container with the default configuration (where the default memory assigned for a VM is 2048 MiB), 
+    # `/run` would be allocated around 200 MiB. Consequently, we can only pull images up to 200 MiB in the guest. 
+    # And the unpacked size of image "ghcr.io/confidential-containers/test-container:big-size" is 965MB. 
+    # It will fail to run the pod with pulling the image in the memory in the guest by default. 
+
+    pod_config="$(new_pod_config "$large_image" "kata-${KATA_HYPERVISOR}")"
+    set_node "$pod_config" "$node"
+    set_container_command "$pod_config" "0" "sleep" "30"
 
     # For debug sake
-    echo "Pod $kata_pod_with_nydus_config file:"
-    cat $kata_pod_with_nydus_config
+    echo "Pod $pod_config file:"
+    cat $pod_config
 
     # The pod should be failed because the default timeout of CreateContainerRequest is 60s
-    assert_pod_fail "$kata_pod_with_nydus_config"
+    assert_pod_fail "$pod_config"
     assert_logs_contain "$node" kata "$node_start_time" \
 		'context deadline exceeded'
 
-    kubectl delete -f $kata_pod_with_nydus_config
+    kubectl delete -f $pod_config
+
+    pod_config="${pod_config_dir}/pod-guest-pull-in-trusted-storage.yaml"
+    storage_config="${pod_config_dir}/confidential/trusted-storage.yaml"
+    local_device=$(create_loop_device "/tmp/trusted-storage.img")
+    sed -i "s/RUNTIMECLASS/kata-${KATA_HYPERVISOR}/g" $pod_config
```

Comment:
This isn't needed

Reply:
What do you mean?

Reply:
The RUNTIMECLASS replacement in the pod.yaml is done automatically if it's `kata`, so we don't need the sed in the test code